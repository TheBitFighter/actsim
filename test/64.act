defproc src (chan(int,int) out)
{
  int a, b;
  chp {
    a := 0;
    b := 0;
   *[ b < 10 -> out!a?b; log ("sent: ", a, " got: ", b); a := a + 2 ]
  }
}

defproc dst (chan(int,int) in)
{
  int a, b;
  chp {
    a := 1; 
    b := 1;
    b := 1;
    b := 1;
    b := 1;
   *[ in?a!b; log ("sent ", b, " got: ", a); b := b + 2 ]
  }
}


defproc test()
{
  src x1;
  dst x2(x1.out);
}
