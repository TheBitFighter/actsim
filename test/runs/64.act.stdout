[                  60] <x2>  sent 1 got: 0
[                  60] <x1>  sent: 0 got: 1
[                  80] <x1>  sent: 2 got: 3
[                  80] <x2>  sent 3 got: 2
[                 100] <x2>  sent 5 got: 4
[                 100] <x1>  sent: 4 got: 5
[                 120] <x1>  sent: 6 got: 7
[                 120] <x2>  sent 7 got: 6
[                 140] <x2>  sent 9 got: 8
[                 140] <x1>  sent: 8 got: 9
[                 160] <x1>  sent: 10 got: 11
[                 160] <x2>  sent 11 got: 10
